_tools/check_browser_compat.ts

```diff
@@ -26,24 +29,24 @@ function isBrowserCompatible(filePath: string): boolean {
   return success;
 }
 
-function hasBrowserCompatibleComment(path: string): boolean {
-  const output = Deno.readTextFileSync(path);
-  return output.includes(DECLARATION);
+function hasBrowserCompatibleComment(source: string): boolean {
+  return source.includes(DECLARATION);
 }
 
-const maybeBrowserCompatibleFiles: string[] = [];
-
-for (const { path } of walkSync(ROOT, { exts: [".ts"], skip: SKIP })) {
-  if (isBrowserCompatible(path) && !hasBrowserCompatibleComment(path)) {
-    maybeBrowserCompatibleFiles.push(path);
+for await (const { path } of walk(ROOT, { exts: [".ts"], skip: SKIP })) {
+  const source = await Deno.readTextFile(path);
+  if (isBrowserCompatible(path) && !hasBrowserCompatibleComment(source)) {
+    if (CHECK) {
+      console.log(
+        `${path} is likely browser-compatible and can have the "${DECLARATION}" comment added.`,
+      );
+    } else {
+      const index = source.indexOf(COPYRIGHT);
+      await Deno.writeTextFile(
+        path,
+        source.slice(0, index + COPYRIGHT.length) + "\n" + DECLARATION + "\n" +
```

Comment:
The second `"\n"` looks redundant to me.

Reply:
This section makes me wonder whether we should integrate both copyright and web compatibility checks. Either way, an idea for another issue/PR.

Reply:
> The second `"\n"` looks redundant to me.

removed

Reply:
> This section makes me wonder whether we should integrate both copyright and web compatibility checks. Either way, an idea for another issue/PR.

I agree, this looks ugly and hacky. I had a try with the ts compiler API which would work without direct string manipulation, but the ts printer doesn't retain format and whitespaces. The same problem appeared on a draft of auto-inserting `warnOnDeprecatedApi()` calls for deprecated functions. Is there a simple ts parser that also retains format and whitespaces?